src/Workspaces/CSharp/Portable/Extensions/ContextQuery/CSharpSyntaxContext.cs

```diff
@@ -283,6 +283,27 @@ public bool IsTypeAttributeContext(CancellationToken cancellationToken)
             return false;
         }
 
+        public bool IsTypeAttributeContextBefore(SyntaxKind syntaxKind, CancellationToken cancellationToken)
```

Comment:
@CyrusNajmabadi This method is better than other I was using? Would you like I change the method name or something in this method?